stub/src/main/java/io/grpc/stub/ServerCalls.java

```diff
@@ -349,4 +349,31 @@ public void onCancel() {
     public void onComplete() {
     }
   }
+
+  public static void unimplementedStreamResponse(StreamObserver responseObserver) {
+    responseObserver.onError(Status.UNIMPLEMENTED.asException());
+  }
+
+  public static class UnimplementedStreamObserver<ReqT> implements StreamObserver<ReqT> {
+    private final StreamObserver responseObserver;
+
+    public UnimplementedStreamObserver(StreamObserver responseObserver) {
+      this.responseObserver = responseObserver;
+    }
+
+    @Override
+    public void onNext(ReqT value) {
+      unimplementedStreamResponse(responseObserver);
+    }
+
+    @Override
+    public void onError(Throwable t) {
+      unimplementedStreamResponse(responseObserver);
+    }
+
+    @Override
+    public void onCompleted() {
+      unimplementedStreamResponse(responseObserver);
```

Comment:
Not sure if calling `unimplementedStreamResponse` in every method is the right thing to do?